tests/external_executor/external_executor.py

```diff
@@ -421,10 +450,50 @@ def compare_op_results(stub, n_ops):
     return network
 
 
+@reqs.description("Test multiple executors that support the same endpoint")
+def test_multiple_executors(network, args):
+    primary, _ = network.find_primary()
+
+    supported_endpoints_a = [
+        ("GET", "/article_description/Monday"),
+        ("POST", "/update_cache/Monday"),
+    ]
+    supported_endpoints_b = [("GET", "/article_description/Monday")]
+
+    executor_a_credentials = register_new_executor(
+        primary, network, supported_endpoints=supported_endpoints_a
+    )
+    executor_b_credentials = register_new_executor(
+        primary, network, supported_endpoints=supported_endpoints_b
+    )
+
+    with executor_thread(WikiCacherExecutor(primary, executor_a_credentials)):
+        with primary.client() as c:
+            r = c.post("/update_cache/Monday")
+            assert r.status_code == http.HTTPStatus.OK
+            content = r.body.text().splitlines()[-1]
+
+            r = c.get("/article_description/Monday")
+            assert r.status_code == http.HTTPStatus.OK
+            assert r.body.text() == content
+
+    # /article_description/Earth this time will be passed to executor_b
+    with executor_thread(WikiCacherExecutor(primary, executor_b_credentials)):
+        with primary.client() as c:
+            r = c.get("/article_description/Monday")
+            assert r.status_code == http.HTTPStatus.OK
+            assert r.body.text() == content
+
+    return network
+
+
 def test_logging_executor(network, args):
     primary, _ = network.find_primary()
 
-    credentials = register_new_executor(primary, network)
+    supported_endpoints = {("POST", "/app/log/public"), ("GET", "/app/log/public")}
```

Comment:
This should be stored in `LoggingExecutor`, something like `LoggingExecutor.supported_endpoints`? The equivalent for `WikiCacher` would need to be a function (taking the list of topics to construct the concrete endpoints), until we support slug templates.